### PR TITLE
rfb: add RFB frames

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://suricata.readthedocs.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/5717
 
Previous PR: #8340 

Describe changes:
- Fix format issues

suricata-verify-pr: 1059

### CODE REVIEW

File: rust/src/rfb/rfb.rs [hunk 1]
```diff
@@ -248,7 +269,11 @@ impl RFBState {
                             consumed += current.len() - rem.len();
                             current = rem;
 
-                            self.state = parser::RFBGlobalState::TCServerInit;
+                        if let Some(frame) = &pdu {
```
Comment (hsadia538) [inline rust/src/rfb/rfb.rs:278]: Same indentation issue here

File: rust/src/rfb/rfb.rs [hunk 2]
```diff
@@ -388,7 +428,11 @@ impl RFBState {
                             consumed += current.len() - rem.len();
                             current = rem;
 
-                            self.state = parser::RFBGlobalState::TSVncResponse;
+                        if let Some(frame) = &pdu {
```
Comment (hsadia538) [inline rust/src/rfb/rfb.rs:458]: same issue

File: rust/src/rfb/rfb.rs [hunk 3]
```diff
@@ -226,7 +243,11 @@ impl RFBState {
                             consumed += current.len() - rem.len();
                             current = rem;
 
-                            self.state = parser::RFBGlobalState::TCSecurityResult;
+                        if let Some(frame) = &pdu {
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply rust/src/rfb/rfb.rs:251]: tab vs spaces?

[the remark this answers is not in the document]
Comment (jufajardini) [reply rust/src/rfb/rfb.rs:251]: We'll try to figure out what's going on with it tomorrow...

**Comment by hsadia538** [comment]
followed by #8362

**Pull request closed without merging**